### ISSUE TITLE
[REQ] F-Droid stuck on 3.0.0-5

### ISSUE THREAD

**Issue by Mrnofish** [jonjomckay/fritter#675]
**Describe the problem that needs solving**
F-Droid only offers beta versions up to 3.0.0-5. Given the amount of problematic changes Twitter has gone through, lately, this is likely to aggravate the problem with bug reports accumulating, from folks who may not check the releases page before hitting the New Issue button.

**Comment by Johndeep** [comment]
After a bit of research as an outsider and someone, who also enjoys using fritter, I assume that the changes has to be made at gitlab in their repository in
fdroiddata  /metadata / com.jonjomckay.fritter.yml
This seems to be used for regular update for Fritter in F-Droid repo. But the last update made by @jonjomckay was 4 month ago.

Is it possible to do own PRs there to help out?

**Comment by Mrnofish** [comment]
Yeah I'm sorry I can't help much here, however if I may think aloud for a while:

I am aware f-droid can pick up new releases automatically and there are new versions on GitHub's so it was puzzling to find out that it was behind, but honestly not sure the exact details of how this works and whether it was a deliberate decision.

Currently old betas appear to be completely broken: beta 5 couldn't load my feed and later wouldn't even load individual accounts, IIRC.

I'm on beta 7 now and haven't seen any show stoppers, unless there are other disasters I haven't noticed (definitely possible), publishing it on f-droid would seem preferable to keeping it on beta 5.

**Comment by jonjomckay** [comment]
I just never got around to it. The F-Droid policies/CI pipelines/detection stuff seems to change every time I publish a new version of Fritter, and I don't really have the free time to wait around for each time the build fails on their side, fix it, and wait again - I only have a small amount of time I'm able to dedicate to Fritter.

I'd recommend using a GitHub build for the time being. It gets built automatically, and is verifiable that it comes directly from the GitHub source.

**Comment by lazytownfan** [comment]
Since this was mentioned, I'll also add my observations: I've noticed that F-Droid thinks that version `2.16.4` is newer than version `3.0.0-beta5`.

Initially, I was trying to figure out why inexplicably I no longer had the option to disable screenshots in Fritter and had initially believed that the feature that reverted #355, but I realized this wasn't the case after cross referencing the GH [releases](https://github.com/jonjomckay/fritter/releases) with the versions on [F-Droid](https://f-droid.org/en/packages/com.jonjomckay.fritter/).

Below is a screenshot from Droid-ify (I'm using Droid-ify, though the same will apply for F-Droid and any other F-Droid client):

![screenshot](https://user-images.githubusercontent.com/93500165/235329157-fc07a564-84f9-4b9b-840d-f4dfb1231927.png)

**Comment by Mrnofish** [comment]
Don't quote me on this, it may be because "stable" versions are prioritized over "unstable" ones. I guess if you're interested in clearing up this matter, it's a question that may be better brought up on the F-Droid forum, though.

**Comment by lazytownfan** [comment]
@Mrnofish Would I basically make a new thread on the F-Droid Forum?

I'm a bit wary of "making yet another online account", though this shouldn't be too bad with a good password manager.  (Also, it's running on Discourse, which offers proper TOTP - i.e., not locked down to some weird closed-sourced always-online concoction like Authy or Duo Security - and security keys for 2FA.)

**Comment by Mrnofish** [comment]
On second thought, if you take a look at Fritter's yaml file in the fdroiddata repo, the order as seen in the app seems the the same as in the file: the 3.0 betas are squeezed between 2.16.3 and 2.16.4.

Peeking into the file history, 2.16.4 was added to Fritter's yaml *after* 3.0.0-beta5, and the timestamps are probably the same you're seeing in the store app.

Anyway, if forums aren't your cuppa, there are both an IRC channel and a Matrix room.

**Comment by jonjomckay** [comment]
I'm struggling with this. I tried [pushing beta9 to F-Droid](https://gitlab.com/jonjomckay/fdroiddata/-/commit/22c80bb72f566c3c918576c7972e58225e4a041b) the other day and the build failed for yet another reason. It always builds fine locally and on GitHub so this is something F-Droid specific.

Unfortunately, the F-Droid build takes SO LONG and I don't currently have the free time to wait around for it to break and fix and break and fix.

If someone fancies taking a look, I'd be very appreciative 🙂

**Comment by shuvashish76** [comment]
v`3.0.0-beta9` now available on F-Droid :)

**Comment by jonjomckay** [comment]
OK, that wasn't me, and it looks like it might not have been done completely as the changelog is missing 😢

I'll see if I can sort it out, but I don't think so because it's based on the Git tag...
